impeller/compiler/shader_lib/impeller/branching.glsl

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef BRANCHING_GLSL_
+#define BRANCHING_GLSL_
+
+#include <impeller/constants.glsl>
+
+vec3 EqualTo3(vec3 x, float y) {
+  return 1 - abs(sign(x - y));
+}
+
+float GreaterThan(float x, float y) {
```

Comment:
Here and elsewhere, some headerdoc comments? Simple ones to start off with are fine.